### ISSUE TITLE
Executing `AnimationTreePlayer.are_nodes_connected` crashes Godot

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#62200]
### Godot version

3.5.beta.custom_build. f880904c7ad9a7a1c808643c2f695126fd5c313f

### System information

Ubuntu 22.04 - Nvidia GTX 970, Gnome shell 42 X11

### Issue description

When executing
```
	var temp_variable17608 = AnimationTreePlayer.new()
	add_child(temp_variable17608)
	temp_variable17608.add_node(3, ".")
	temp_variable17608.add_node(2, "")
	temp_variable17608.are_nodes_connected("", ".", -2)
```

Godot crashes with this backtrace
```
ERROR: FATAL: Index p_index = -2 is out of bounds (size() = 2).
   at: get (./core/cowdata.h:149)

================================================================
handle_crash: Program crashed with signal 4
Engine version: Godot Engine v3.5.rc.custom_build (f880904c7ad9a7a1c808643c2f695126fd5c313f)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7fc4bcf37520] (??:0)
[2] CowData<AnimationTreePlayer::Input>::get(int) const (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/cowdata.h:149 (discriminator 7))
[3] Vector<AnimationTreePlayer::Input>::operator[](int) const (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/vector.h:88)
[4] AnimationTreePlayer::are_nodes_connected(StringName const&, StringName const&, int) const (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/animation/animation_tree_player.cpp:1403)
[5] MethodBind3RC<bool, StringName const&, StringName const&, int>::call(Object*, Variant const**, int, Variant::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/method_bind.gen.inc:2849 (discriminator 16))
[6] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object.cpp:918 (discriminator 1))
[7] Variant::call_ptr(StringName const&, Variant const**, int, Variant*, Variant::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/variant_call.cpp:1221 (discriminator 1))
[8] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript_function.cpp:1053)
[9] GDScriptInstance::call_multilevel(StringName const&, Variant const**, int) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript.cpp:1211)
[10] Node::_notification(int) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/node.cpp:56)
[11] Node::_notificationv(int, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./scene/main/node.h:47 (discriminator 14))
[12] Object::notification(int, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object.cpp:929)
[13] SceneTree::_notify_group_pause(StringName const&, int) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:1105)
[14] SceneTree::idle(float) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:608 (discriminator 3))
[15] Main::iteration() (/home/runner/work/GodotBuilds/GodotBuilds/godot/main/main.cpp:2286)
[16] OS_X11::run() (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/x11/os_x11.cpp:3933)
[17] godot(main+0x125) [0x185db0b] (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/x11/godot_x11.cpp:57)
[18] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7fc4bcf1ed90] (??:0)
[19] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7fc4bcf1ee40] (??:0)
[20] godot(_start+0x2e) [0x185d92e] (??:?)
-- END OF BACKTRACE --
================================================================

```

This example was found by Godot fuzzer - [Qarminer](https://github.com/qarmin/Qarminer), so it is quite unlikelly that this code could be used in real project, but still this should be handled gracefully.

### Steps to reproduce

Above

**Comment by Anutrix** [comment]
```ERR_FAIL_INDEX(p_dst_input, dst->inputs.size());```
Adding this on the line before return statement in `are_nodes_connected` function seems to be the fix.

**Comment by akien-mga** [comment]
Fixed by #63783.
